src/resty/aws/request/build.lua

```diff
@@ -106,7 +106,7 @@ local function build_request(operation, config, params)
   local body_tbl = {}
 
   if config.signingName or config.targetPrefix then
-    request.headers["X-Amz-Target"] = (config.signingName or config.targetPrefix) .. "." .. operation.name
+    request.headers["X-Amz-Target"] = (config.targetPrefix or config.signingName) .. "." .. operation.name
```

Comment:
the `x-amz-target` should have the `Comprehend_20171127` prefix. In this case `targetPrefix` has the `Comprehend_20171127` value.

Reply:
Will this affect other APIs?